src/main/c/openssl/extconf.rb

```diff
@@ -30,11 +30,10 @@
 
 if ENV['OPENSSL_PREFIX']
   $CFLAGS += " -I #{ENV['OPENSSL_PREFIX']}/include"
-  $LIBS += " -l #{ENV['OPENSSL_PREFIX']}/lib/libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
-else
-  $LIBS += " -l libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
 end
 
+$LIBS += " -l libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
```

Comment:
I think it's good to combine both and only give a prefix-like option, as then it's impossible to pick the headers from one prefix and the lib from another.